packages/shylu/shylu_dd/frosch/src/SchwarzOperators/FROSch_CoarseOperator_def.hpp

```diff
@@ -231,17 +231,30 @@ namespace FROSch {
             //------------------------------------------------------------------------------------------------------------------------
             // Matrix to the new communicator
             if (OnCoarseSolveComm_) {
-                CoarseMatrix_ = Xpetra::MatrixFactory<SC,LO,GO,NO>::Build(CoarseSolveMap_,k0->getGlobalMaxNumRowEntries());
+                LO numRows = tmpCoarseMatrix->getNodeNumRows();
+                Teuchos::ArrayRCP<size_t> elemsPerRow(numRows);
                 ConstGOVecView indices;
                 ConstSCVecView values;
-                for (UN i=0; i<tmpCoarseMatrix->getNodeNumRows(); i++) {
-                    tmpCoarseMatrix->getGlobalRowView(CoarseSolveMap_->getGlobalElement(i),indices,values);
+                for (LO i = 0; i < numRows; i++) {
+                  GO globalRow = CoarseSolveMap_->getGlobalElement(i);
+                  size_t numEntries = tmpCoarseMatrix->getNumEntriesInGlobalRow(globalRow);
+                  if(numEntries == 0)
+                  {
+                    //Always add the diagonal for empty rows
+                    numEntries = 1;
+                  }
+                  elemsPerRow[i] = numEntries;
+                }
+                CoarseMatrix_ = Xpetra::MatrixFactory<SC,LO,GO,NO>::Build(CoarseSolveMap_, elemsPerRow, Xpetra::StaticProfile);
```

Comment:
Does this invoke a deprecated constructor that takes ArrayRCP instead of ArrayView?  Perhaps it's not deprecated for Xpetra.

Reply:
It's the only constructor for xpetra. Might as well switch over because arrayRCP has operator() to get a view.